Documentation/operations/upgrade.rst

```diff
@@ -307,7 +307,10 @@ Annotations:
 1.16 Upgrade Notes
 ------------------
 
-* To be determined during v1.16 development.
+* Cilium Envoy daemon set is now enabled by default. If you are upgrading from a
```

Comment:
Have you experimented with the `upgradeCompatibility` Helm value? In other words, if users set `upgradeCompatitibilty <= 1.15" (which tells us they are upgrading from an existing Cilium version), is there a way to disable the DaemonSet?

This is typically how we deal with breaking changes in Helm values. Most of them only apply to ConfigMaps, but depending on the complexity it might also be possible to use here:
https://github.com/cilium/cilium/blob/7318ce2d0d89a91227e3f313adebce892f3c388e/install/kubernetes/cilium/templates/cilium-configmap.yaml#L2-L85

Reply:
Yes, I looked at upgradeCompatibility flag before, but it seems to work for non-default attribute only, for example,   kubeProxyReplacement is commented in values.yaml.